examples/table_script/table_script.py

```diff
@@ -0,0 +1,198 @@
+"""
+Draft script to train a simple sklearn model 
+and output the results that could be used to create 
+a table similar to what is from the paper.
+
+"""
+
+from pprint import pprint
+
+import numpy as np
+import pandas as pd
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem
+from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
+from sklearn.svm import LinearSVR
+
+from astartes import train_val_test_split
+
+# read in the data
+CSV_PATH = '../barrier_prediction_with_RDB7/ccsdtf12_dz.csv'
+df = pd.read_csv(CSV_PATH)
+print(df.shape)
+
+# helper function to featurize the data with 2048 morgan fingerprint
+# https://github.com/chemprop/chemprop/blob/master/chemprop/features/features_generators.py
+MORGAN_RADIUS = 2
+MORGAN_NUM_BITS = 2048
+def morgan_counts_features_generator(mol,
+                                     radius= MORGAN_RADIUS,
+                                     num_bits= MORGAN_NUM_BITS):
+    """
+    Generates a counts-based Morgan fingerprint for a molecule.
+    :param mol: A molecule (i.e., either a SMILES or an RDKit molecule).
+    :param radius: Morgan fingerprint radius.
+    :param num_bits: Number of bits in Morgan fingerprint.
+    :return: A 1D numpy array containing the counts-based Morgan fingerprint.
+    """
+    mol = Chem.MolFromSmiles(mol) if type(mol) == str else mol
+    features_vec = AllChem.GetHashedMorganFingerprint(mol, radius, nBits=num_bits)
+    features = np.zeros((1,))
+    DataStructs.ConvertToNumpyArray(features_vec, features)
+
+    return features
+
+# create X and y
+params = Chem.SmilesParserParams()
+params.removeHs = False
+
+X = np.zeros((len(df), 2048*2))
+for i, row in df.iterrows():
+    rsmi, psmi = row.rsmi, row.psmi
+
+    rmol = Chem.MolFromSmiles(rsmi, params)
+    r_morgan = morgan_counts_features_generator(rmol)
+
+    pmol = Chem.MolFromSmiles(psmi, params)
+    p_morgan = morgan_counts_features_generator(pmol)
+
+    X[i, :] = np.concatenate((r_morgan,
+                              p_morgan - r_morgan),
+                             axis=0)
+
+y = df.dE0.values
+
+def produce_table(sklearn_model,
```

Comment:
I like this signature, much less error prone than what I had envisioned. My only bone to pick would be with the name `sampler_hopts`. This suggestion sounds dumb, but I think it should be more like `samplers_hopts` (with an s on sampler) to indicate that it is a nested dictionary. Perhaps we could change the name entirely? A type hint?

Reply:
Thanks! That's a good suggestion. It's important for variable names to clearly communicate what they are. I also renamed function to `generate_results_dict` since that is its main objective.